articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,455 @@
+```qsharp
+    operation MeasureSuperpositionArray(n : Int) : Result[] {
+        mutable resultArray = new Result[n];     // instantiate a length n array for the measurement Results 
+        using (qubits = Qubit[n]) {              // allocate a register of n qubits
+            ApplyToEach( H , qubits);            // apply H to each qubit in the register (from Microsoft.Quantum.Canon)
+            set resultArray = MultiM(qubits);    // measure the array of qubits
+            ResetAll(qubits);                    // reset all the qubits
+        }
+        return resultArray;
+    }
+```
+where the returned value is an array of the measurement results.
+Note that the [`ApplyToEach`](xref:microsoft.quantum.canon.applytoeach) operation is in the `Microsoft.Quantum.Canon` namespace, requiring another `open` statement with the others.
+
+If we move the `@EntryPoint()` to precede this new operation (note there can only be one such line in a file), attempting to run it with simply `dotnet run` results in a helpful error message.
+This message indicates what additional command line options are required, and how to express them.
+
+The general format for the command line is actually `dotnet run [options]`, and operation arguments are provided there.
+In this case, the argument `n` is missing, and it shows that we need to provide the option `-n <n>`. 
+To run `MeasureSuperpositionArray` for `n=4` qubits, we therefore use
+
+```bash
+dotnet run -n 4
+```
+
+yielding an output similar to
+
+```bash
+[Zero,One,One,One]
+```
+
+This of course extends to multiple arguments.
+
+The error message also provides other options which can be used, including how to change the target machine.
+
+### Different target machines
+
+From the output of our operation thus far, it's clear that the default target machine from the command line is the full-state quauntum simulator, `QuantumSimulator`.
+However, we can instruct the operation to be run on a specific target machine with the option `--simulator` (or the shorthand `-s`).
+
+For example, we could run it on [`ResourcesEstimator`](xref:microsoft.quantum.machines.resources-estimator):
+
+```bash
+dotnet run -n 4 -s ResourcesEstimator
+```
+
+The printed output is then
+
+```bash
+Metric          Sum
+CNOT            0
+QubitClifford   4
+R               0
+Measure         8
+T               0
+Depth           0
+Width           4
+BorrowedWidth   0
+```
+
+Note that the [`Reset`](xref:microsoft.quantum.intrinsic.reset) and [`ResetAll`](xref:microsoft.quantum.intrinsic.resetall) operations utilize measurement to reset the qubits to $\ket{0}$, hence the reported 8 measurements as opposed to the 4 which yield the actual results.
+Details on what these metrics indicate can be found [here](xref:microsoft.quantum.machines.resources-estimator#metrics-reported).
+
+### Command line execution summary
+<br/>
+<img src="./images/command_line_diagram.png" alt="Q# program from command line" width="700">
```

Comment:
File I/O is intentionally out of scope for Q#; if you want to create a file for storing results, the best way to do so with Q# standalone would be to use shell redirection (`dotnet run > foo.txt`).

Reply:
@geduardo I was toying with that as well. I just added a note earlier in the page mentioning just, but I think explicitly including it in the diagrams might risk cluttering them more than necessary. Both because that behavior (printing to the respective console) is a commonality across all the execution methods, and because it's a relatively minor case restricted to those few operations/functions. Let me know what you think when I add the new commits (i.e. if the note at the beginning is sufficient or not)